### PR TITLE
[CAMEL-18612] Inconsistency in JsonPath component causes problems with databinding

### PR DESCRIPTION
**Pull request by rnetuka**
Issue: https://issues.apache.org/jira/browse/CAMEL-18612

Please, check that I didn't miss any place to inlude/support the new _unpackArray_ option.

Also, the option name is only a proposal. Another proposals are:
- unwrapSingleArray (original)
- unwrapArray
- unwrapSingleElements
- unpackSingleArray
- unpackArray
- unpackSingleElements

Which is the best?

### CODE REVIEW

File: components/camel-jsonpath/src/main/java/org/apache/camel/jsonpath/JsonPathLanguage.java [hunk 1]
```diff
@@ -134,8 +140,9 @@ public Expression createExpression(String expression, Object[] properties) {
         answer.setAllowSimple(property(boolean.class, properties, 2, allowSimple));
         answer.setAllowEasyPredicate(property(boolean.class, properties, 3, allowEasyPredicate));
         answer.setWriteAsString(property(boolean.class, properties, 4, writeAsString));
-        answer.setHeaderName(property(String.class, properties, 5, headerName));
-        String option = (String) properties[6];
+        answer.setWriteAsString(property(boolean.class, properties, 5, unpackArray));
```
Comment (davsclaus) [inline components/camel-jsonpath/src/main/java/org/apache/camel/jsonpath/JsonPathLanguage.java:143]: setUnpackArray

**Review by davsclaus — CHANGES_REQUESTED** [review]
We also need to add this new option in 
https://github.com/apache/camel/blob/main/core/camel-core-model/src/main/java/org/apache/camel/model/language/JsonPathExpression.java

**Comment by rnetuka** [comment]
@davsclaus What about https://github.com/apache/camel/blob/main/core/camel-core-model/src/main/java/org/apache/camel/builder/ExpressionClause.java#L466 and  https://github.com/apache/camel/blob/main/core/camel-core-model/src/main/java/org/apache/camel/builder/ExpressionClauseSupport.java#L477?

**Review by davsclaus — APPROVED** [review]

**Comment by davsclaus** [comment]
Yeah ideally we need a new method in those java-dsl classes to expose this new option. However its likely seldom in use, so you can just add 1 new method.

**Pull request merged**